### DISCUSSION TITLE
How to configure Key on Linux and aspnet core

### DISCUSSION THREAD

**Discussion by maxima120** [dotnet/aspnetcore#28450] [Q&A]
I have been kicked around. From aspnet github to stackoverflow to identityserver github and back to aspnet..
Why would you roll out a product without documentation and support.. Alice in Wonderland 2020.

Anyone can help at all? 

### Question

How do I configure Key on Linux?

The following config works fine on Windows. How do I do the same on Linux (CentOS)?
There is no `certificate store` is there?

### Minimal working example

```json
  "IdentityServer": {
    "Clients": {
      "HelloLinuxAuth": {
        "Profile": "IdentityServerSPA"
      }
    },
    "Key": {
      "Type": "Store",
      "StoreName": "My",
      "StoreLocation": "LocalMachine",
      "Name": "CN=my-demo"
    }
  },
```

**Comment by engineering87** [comment]
Hello there,
on Linux there is no Windows-like certificate store such "LocalMachine." Instead, certificates are typically stored as files on the filesystem. To configure IdentityServer to use a certificate on Linux, you'll need to adjust the configuration to use the certificate from a file for example:

```json
"IdentityServer": {
  "Clients": {
    "HelloLinuxAuth": {
      "Profile": "IdentityServerSPA"
    }
  },
  "Key": {
    "Type": "File",
    "FilePath": "/path/to/your/certificate.pfx",
    "Password": "your-password-if-applicable"
  }
}
```

Also ensure that your IdentityServer configuration in `Startup.cs` loads the certificate correctly:

```csharp
// ...
var cert = new X509Certificate2(Path.Combine(Environment.ContentRootPath, "certificate.pfx"), "your-password");
builder.AddSigningCredential(cert);
```
